src/store.js

```diff
@@ -1,8 +1,104 @@
-import {createStore} from 'redux'
+import { createStore } from 'redux'
+
+import * as types from './constants/actionTypes';
+
+const initialState = {
+  policies: [],
+  policyTypes: [],
+  updateForms: []
+};
 
 // add reducers here
-const reducer = ()=>{
+const reducer = (state = initialState, action) => {
+  let policies;
+  let policyTypes;
+  let updateForms;
+
+  switch (action.type) {
+    case types.GET_POLICIES: {
+      console.log('In GET_POLICIES action');
+      return state;
+    }
+
+    case types.ADD_POLICY: {
+      console.log('In ADD_POLICY action');
+      const newPolicy = action.payload;
+      policies = state.policies.slice();
+      updateForms = state.updateForms.slice();
+      policies.push(newPolicy);
+      updateForms.push({
+        policyNumber: newPolicy.policyNumber,
+        renderForm: false,
+        selectedType: null
+      })
+      return {
+        ...state,
+        policies,
+        updateForms
+      }
+    }
+
+    case types.ADD_POLICY_TYPE: {
+      console.log('In ADD_POLICY_TYPE action');
+      policyTypes = state.policyTypes.slice();
+      policyTypes.push(action.payload);
```

Comment:
With this approach, you are keeping two copies of the policyTypes - one in the policies array and another in the policyTypes array. If you had more time, would you have done this differently?

Reply:
My understanding was that we wanted to keep both the policies array and the policyTypes array (both from `provider.js`) in the store. The policyTypes array in store felt redundant/unnecessary to me in general, but the prompt instructed me to import all the policy types from the policyTypes array in `provider.js`:

```Both the policies array retrieved from getPolicies and the policyTypes array from getPolicyTypes should be stored in the redux store.```

I may be misunderstanding your question, though!